src/figuretype/cartpusher.c

```diff
@@ -324,6 +338,14 @@ void figure_cartpusher_action(figure *f)
             }
             f->image_offset = 0;
             break;
+        case FIGURE_ACTION_238_CARTPUSHER_WAITING_FOR_DESTINATION:
+            set_cart_graphic(f, 1);
+            f->wait_ticks++;
+            if (f->wait_ticks > NON_STORABLE_RESOURCE_CARTPUSHER_MAX_WAIT_TICKS) {
+                f->state = FIGURE_STATE_DEAD;
```

Comment:
Gets encased in concrete? :)

Reply:
My reasoning is that concrete dries up if you wait too long